geckomat/change_model/findMetSmiles.m

```diff
@@ -0,0 +1,35 @@
+function metSmiles = findMetSmiles(metNames)
+% findMetSMILES
+%   Queries PubChem by metabolite names to obtain SMILES.
+%
+% Input:
+%   metNames    array with metabolite names (e.g. model.metNames)
+%
+% Ouput:
+%   metSmiles   array with matching SMILES. If model.metNames is used as
+%               input, metSmiles can be specified as model.metSmiles.
+% 
+
+[uniqueNames, ~, uniqueIdx]          = unique(metNames);
+uniqueSmiles(1:numel(uniqueNames),1) = {''};
+
+fprintf('Querying PubChem for SMILES by metabolite names...   0%% complete');
+numUnique = numel(uniqueNames);
+for i = 1:numel(uniqueNames)
+    if rem(i-1,floor(numUnique/100+1)) == 0
+        progress = num2str(floor(100*(i/numUnique)));
+        progress = pad(progress,3,'left');
+        fprintf('\b\b\b\b\b\b\b\b\b\b\b\b\b%s%% complete',progress);
+    end
+        try
+            smileResult       = webread(['https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/name/' uniqueNames{i} '/property/CanonicalSMILES/TXT']);
```

Comment:
When PubCHEM returns multiple SMILES for one compound. We should only keep the first one using `strsplit`, otherwise the output file will be a mass.


#### Expected behavior:

Only one SMILES will be documented for one compound


#### Reproducing these results:

```matlab
% here is one example
`strtrim(webread(['https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/name/3alpha,7alpha-dihydroxy-5beta-cholest-24-enoyl-CoA/property/CanonicalSMILES/TXT']))

% which will return four smiles
 'CC(CCC=C(C)C(=O)SCCNC(=O)CCNC(=O)C(C(C)(C)COP(=O)(O)OP(=O)(O)OCC1C(C(C(O1)N2C=NC3=C(N=CN=C32)N)O)OP(=O)(O)O)O)C4CCC5C4(CCC6C5C(CC7C6(CCC(C7)O)C)O)C
     CC(CCC=C(C)C(=O)SCCNC(=O)CCNC(=O)C(C(C)(C)COP(=O)(O)OP(=O)(O)OCC1C(C(C(O1)N2C=NC3=C(N=CN=C32)N)O)OP(=O)(O)O)O)C4CCC5C4(CCC6C5C(CC7C6(CCC(C7)O)C)O)C
     CC(CCC=C(C)C(=O)SCCNC(=O)CCNC(=O)C(C(C)(C)COP(=O)(O)OP(=O)(O)OCC1C(C(C(O1)N2C=NC3=C(N=CN=C32)N)O)OP(=O)(O)O)O)C4CCC5C4(CCC6C5C(CC7C6(CCC(C7)O)C)O)C
     CC(CCC=C(C)C(=O)SCCNC(=O)CCNC(=O)C(C(C)(C)COP(=O)(O)OP(=O)(O)OCC1C(C(C(O1)N2C=NC3=C(N=CN=C32)N)O)OP(=O)(O)O)O)C4CCC5C4(CCC6C5C(CC7C6(CCC(C7)O)C)O)C
     CC(CCC=C(C)C(=O)SCCNC(=O)CCNC(=O)C(C(C)(C)COP(=O)(O)OP(=O)(O)OCC1C(C(C(O1)N2C=NC3=C(N=CN=C32)N)O)OP(=O)(O)O)O)C4CCC5C4(CCC6C5C(CC7C6(CCC(C7)O)C)O)C'

Reply:
Is this not resolved with the code in line 28?

Reply:
Yes, Sorry for this. I think we used a old version.